src/main/java/use_cases/attack_use_case/AttackInteractor.java

```diff
@@ -0,0 +1,121 @@
+        CreatureCard creatureAttacker = player1.getCreatureById(inputData.getAttackerId());
+        CreatureCard creatureDefender = player2.getCreatureById(inputData.getTargetId());
+        player1.spendEssence(creatureAttacker.getAttackCost());
+        return getFinishAttackResponseModel(player1, player2, creatureAttacker, creatureDefender);
+
+    }
+
+    @Override
+    public FinishAttackResponseModel defend(AttackRequestModel inputData) {
+        Player player1 = gameState.getCurrentPlayer();
+        Player player2 = gameState.getOpposingPlayer();
+        CreatureCard creatureAttacker = player1.getCreatureById(inputData.getAttackerId());
+        CreatureCard creatureDefender = player2.getCreatureById(inputData.getTargetId());
+
+        player1.spendEssence(creatureAttacker.getAttackCost());
+        player2.spendEssence(creatureDefender.getDefendCost());
+        return getFinishAttackResponseModel(player1, player2, creatureAttacker, creatureDefender);
+
+    }
+
+    private FinishAttackResponseModel getFinishAttackResponseModel(Player player1, Player player2, CreatureCard
+            creatureAttacker, CreatureCard creatureDefender) {
+
+        creatureDefender.takeDamage(creatureAttacker.getAttackDamage());
+
+        List<List<Integer>> player1HitIdHand= getPlayerHitCreatureHandID(player1);
+        List<List<Integer>> player2HitIdHand= getPlayerHitCreatureHandID(player2);
+
+        List<Integer> player1CreatureHitpoints = player1HitIdHand.get(0);
+        List<Integer> player2CreatureHitpoints = player2HitIdHand.get(0);
+        List<Integer>  player1CreatureIds = player1HitIdHand.get(1);
+        List<Integer> player2CreatureIds = player1HitIdHand.get(1);
+        List <Integer> player1HandIds = player1HitIdHand.get(2);
+        List <Integer> player2HandIds = player1HitIdHand.get(2);
+
+        FinishAttackResponseModel updatedModel = new FinishAttackResponseModel(player1CreatureHitpoints,
+                player2CreatureHitpoints, player1CreatureIds, player2CreatureIds, player1HandIds, player2HandIds);
+
+        return attackPresenter.exitDefendInputScreen(updatedModel);
+    }
+
+    private List<List<Integer>> getPlayerHitCreatureHandID(Player player){
+        List<List<Integer>> playerOutput = new ArrayList<>();
+
+        List<Integer> playerCreatureHitpoints = new ArrayList<>();
+        List<Integer> playerCreatureIds= new ArrayList<>();
+        List<Integer> playerHandIds = new ArrayList<>();
+
+        for (int i = 0; i < 3; i ++){
+            playerCreatureHitpoints.add(player.getCreatures().get(i).getHitPoints());
+            playerCreatureIds.add(player.getCreatures().get(i).getId());
+        }
+        for (Card player1Card: player.getHand()) {
+            playerHandIds.add(player1Card.getId());
+        }
+
+        playerOutput.add(playerCreatureHitpoints);
+        playerOutput.add(playerCreatureIds);
+        playerOutput.add(playerHandIds);
+
+        return playerOutput;
+    }
```

Comment:
I get what this method is doing, but we can also just avoid making a big helper method by using a few for loops in the original method. (We're using for loops anyway, so might as well make it in the same method so that the reader doesn't have to jump around places to see the full picture).
```suggestion
```